### PR TITLE
stubtest_stdlib: get rid of --ignore-missing-stub

### LINKED ISSUE #6401: stubtest: disable --ignore-missing-stubs for stdlib?
@AlexWaygood has recently fixed lots of things in the standard library by running `stubtest_stdlib` without the `--ignore-missing-stubs` option. Maybe our CI should do that too, so that we notice what's missing? This doesn't apply to third-party stubs very well, because then our CI would be broken every time a new feature is added to any library, but the standard library is relatively stable.

### PR DESCRIPTION
**Pull request by Akuli**
Fixes #6401.

### CODE REVIEW

**Comment by Akuli** [comment]
This is easiest to review commit by commit.

There are a lot of new entries, but I don't think combining them with wildcards, e.g. `distutils..+`, is a good idea, because that would hide all errors from distutils, not just the errors where something is missing from stubs.

File: tests/stubtest_allowlists/py310.txt [hunk 1]
```diff
@@ -135,3 +135,84 @@ ssl._create_unverified_context
...
+multiprocessing.managers.SharedMemoryServer.public
+multiprocessing.managers.SharedMemoryServer.release_segment
+multiprocessing.managers.SharedMemoryServer.shutdown
+multiprocessing.managers.SharedMemoryServer.track_segment
+pstats.FunctionProfile
+pstats.Stats.get_stats_profile
+pstats.StatsProfile
+pyexpat.XMLParserType.SkippedEntityHandler
+pyexpat.XMLParserType.intern
+re.Pattern.scanner
```
Comment (AlexWaygood) [inline tests/stubtest_allowlists/py310.txt:193]: I think this should be moved to a "Won't fix" section, with a ref to #6405

Comment (AlexWaygood) [inline tests/stubtest_allowlists/py310.txt:193]: I think this should be moved to a "Won't fix" section, with a ref to #6405

File: tests/stubtest_allowlists/py310.txt [hunk 2]
```diff
@@ -135,3 +135,84 @@ ssl._create_unverified_context
...
+sched.Event.sequence
+types.CoroutineType.cr_origin
+types.GenericAlias.__call__
```
Comment (AlexWaygood) [inline tests/stubtest_allowlists/py310.txt:196]: I would move this to the "Won't fix" section as well, with a ref to #6392

File: tests/stubtest_allowlists/py310.txt [hunk 3]
```diff
@@ -135,3 +135,84 @@ ssl._create_unverified_context
 typing.SupportsRound.__round__
```
Comment (AlexWaygood) [inline tests/stubtest_allowlists/py310.txt:135]: While we're here, could we move this to the correct section? I added it in one of my first PRs to typeshed (#6242), and put it in the wrong section.

File: tests/stubtest_allowlists/py37.txt [hunk 4]
```diff
@@ -77,3 +77,62 @@ collections.Generator.gi_running
...
+dummy_threading.setprofile
+dummy_threading.settrace
+dummy_threading.stack_size
+html.parser.HTMLParser.unescape
+importlib.util.source_hash
+lib2to3.pgen2.tokenize.COLONEQUAL
+platform.popen
+plistlib.Data.asBase64
+plistlib.Data.fromBase64
+re.Pattern.scanner
```
Comment (AlexWaygood) [inline tests/stubtest_allowlists/py37.txt:125]: Same as in the py310 entry, I'd put this in a "Won't fix" section with a ref to #6405

File: tests/stubtest_allowlists/py38.txt [hunk 5]
```diff
@@ -96,3 +96,92 @@ collections.Generator.gi_running
...
+multiprocessing.managers.SharedMemoryServer.list_segments
+multiprocessing.managers.SharedMemoryServer.public
+multiprocessing.managers.SharedMemoryServer.release_segment
+multiprocessing.managers.SharedMemoryServer.shutdown
+multiprocessing.managers.SharedMemoryServer.track_segment
+plistlib.Data.asBase64
+plistlib.Data.fromBase64
+pyexpat.XMLParserType.SkippedEntityHandler
+pyexpat.XMLParserType.intern
+re.Pattern.scanner
```
Comment (AlexWaygood) [inline tests/stubtest_allowlists/py38.txt:161]: Same as in the py310 entry, I'd put this in a "Won't fix" section with a ref to #6405

File: tests/stubtest_allowlists/py39.txt [hunk 6]
```diff
@@ -106,3 +106,83 @@ collections.Generator.gi_running
...
+multiprocessing.managers.SharedMemoryServer.public
+multiprocessing.managers.SharedMemoryServer.release_segment
+multiprocessing.managers.SharedMemoryServer.shutdown
+multiprocessing.managers.SharedMemoryServer.track_segment
+pstats.FunctionProfile
+pstats.Stats.get_stats_profile
+pstats.StatsProfile
+pyexpat.XMLParserType.SkippedEntityHandler
+pyexpat.XMLParserType.intern
+re.Pattern.scanner
```
Comment (AlexWaygood) [inline tests/stubtest_allowlists/py39.txt:165]: Same as in the py310 entry, I'd put this in a "Won't fix" section with a ref to #6405

File: tests/stubtest_allowlists/py39.txt [hunk 7]
```diff
@@ -106,3 +106,83 @@ collections.Generator.gi_running
...
+types.CoroutineType.cr_origin
+types.GenericAlias.__call__
```
Comment (AlexWaygood) [inline tests/stubtest_allowlists/py39.txt:167]: Same as in the py310 entry, I'd put this in a "Won't fix" section, with a ref to #6392

File: tests/stubtest_allowlists/py3_common.txt [hunk 8]
```diff
@@ -272,3 +272,437 @@ os.O_[A-Z_]+
...
+codecs.BOM32_LE
+codecs.BOM64_BE
+codecs.BOM64_LE
+codecs.StreamReader.charbuffertype
+codecs.StreamReader.seek
+codecs.StreamWriter.seek
+codecs.namereplace_errors
+configparser.ParsingError.filename
+configparser.RawConfigParser.converters
+csv.__version__
```
Comment (AlexWaygood) [inline tests/stubtest_allowlists/py3_common.txt:321]: I would put this in a "Wont't fix" section, with a ref to #6398

File: tests/stubtest_allowlists/py3_common.txt [hunk 9]
```diff
@@ -272,3 +272,437 @@ os.O_[A-Z_]+
...
+profile.Profile.trace_dispatch_c_call
+profile.Profile.trace_dispatch_call
+profile.Profile.trace_dispatch_exception
+profile.Profile.trace_dispatch_i
+profile.Profile.trace_dispatch_l
+profile.Profile.trace_dispatch_mac
+profile.Profile.trace_dispatch_return
+pydoc.Helper.symbol
+pydoc.Helper.symbols_
+pydoc.Helper.topic
```
Comment (AlexWaygood) [inline tests/stubtest_allowlists/py3_common.txt:608]: We don't want these ever added to the stub, as they're just throwaway variables that are a side-effect of a for-loop (https://github.com/python/typeshed/issues/6401#issuecomment-981178522) -- I'd put them in a "Won't fix" section

File: tests/stubtest_allowlists/py3_common.txt [hunk 10]
```diff
@@ -272,3 +272,437 @@ os.O_[A-Z_]+
...
+random.Random.VERSION
+rlcompleter.Completer.attr_matches
+rlcompleter.Completer.global_matches
+sunau.Au_read.initfp
+sunau.Au_write.initfp
+threading.Event.isSet
```
Comment (AlexWaygood) [inline tests/stubtest_allowlists/py3_common.txt:614]: This is a deprecated alias to another method (https://docs.python.org/3/library/threading.html#threading.Event.is_set), so shouldn't ever be added to the stub in my opinion -- I'd put it in a "Won't fix" section

File: tests/stubtest_allowlists/py3_common.txt [hunk 11]
```diff
@@ -272,3 +272,437 @@ os.O_[A-Z_]+
...
+email.policy.Policy.message_factory
+email.quoprimime
+email.utils
+hmac.HMAC.blocksize
+importlib.abc.Finder.find_module
+importlib.machinery.WindowsRegistryFinder.DEBUG_BUILD
+importlib.machinery.WindowsRegistryFinder.REGISTRY_KEY
+importlib.machinery.WindowsRegistryFinder.REGISTRY_KEY_DEBUG
+inspect.Signature.from_builtin
+inspect.Signature.from_function
```
Comment (AlexWaygood) [inline tests/stubtest_allowlists/py3_common.txt:536]: These functions are deprecated and have been removed in Python 3.11, so we might want to add a comment to that effect or put them in a "Won't fix" section

File: tests/stubtest_allowlists/py3_common.txt [hunk 12]
```diff
@@ -272,3 +272,437 @@ os.O_[A-Z_]+
...
+_thread.allocate
+_thread.exit_thread
+_thread.start_new
+_threading_local._localimpl.localargs
+_threading_local._localimpl.locallock
+asynchat.async_chat.encoding
+asynchat.async_chat.use_encoding
+asynchat.find_prefix_at_end
+asyncore.dispatcher.addr
+asyncore.dispatcher.handle_accepted
```
Comment (AlexWaygood) [inline tests/stubtest_allowlists/py3_common.txt:296]: These modules are deprecated in their entirety, and have been removed entirely in the current main branch of Python. That might be reverted before Python 3.11 is released, but we might want to note that here, or move them to a "Won't fix" section

**Review by AlexWaygood — COMMENTED** [review]
Thanks, this looks great! A few thoughts:

File: tests/stubtest_allowlists/py3_common.txt (hunk 10 above)
Comment (Akuli) [reply tests/stubtest_allowlists/py3_common.txt:614]: threading has other aliases too. We can remove them, or add `isSet`, but either way it's beyond the scope of this PR.

Comment (AlexWaygood) [reply tests/stubtest_allowlists/py3_common.txt:614]: Fair!

**Review by AlexWaygood — APPROVED** [review]

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**